File: moderate/array_absurdity.rb
File.open(ARGV[0]).each_line do |line|
  s = line.split(';')
  next if s.empty?
  m = {}
  s[1].split(',').map(&:to_i).each do |i|
    if m[i]
      puts i
      break
    end
    m[i] = true
  end
end

File: moderate/array_absurdity.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp($line);
    $line =~ s/^[0-9]+;//g;
    my @a;
    foreach ( split( /,/, $line ) ) {
        if ( @a[$_] ) {
            print "$_\n";
            last;
        }
        @a[$_] = 1;
    }
}
close(INFILE);
